orange_cb_recsys/recsys/ranking_algorithms/page_rank.py

```diff
@@ -11,85 +12,311 @@
         super().__init__()
-        self.__personalized = personalized
-        self.__fullgraph: FullGraph = graph
+        self.__fullgraph = fullgraph
+        self.__remove_user_nodes: bool = remove_user_nodes
+        self.__remove_items_in_profile: bool = remove_items_in_profile
+        self.__remove_properties: bool = remove_properties
 
     @property
-    def fullgraph(self):
+    def fullgraph(self) -> FullGraph:
+        """
+        Getter for the graph on which the PageRank algorithm will be executed
+        """
         return self.__fullgraph
 
-    def set_fullgraph(self, graph):
+    @fullgraph.setter
+    def fullgraph(self, graph: FullGraph):
+        """
+        Sets the FullGraph on which the PageRank algorithm will be executed
+        """
         self.__fullgraph = graph
 
     @property
-    def personalized(self):
-        return self.__personalized
+    def remove_user_nodes(self) -> bool:
+        """
+        Getter for the boolean attribute remove_user_nodes
+        """
+        return self.__remove_user_nodes
+
+    @remove_user_nodes.setter
+    def remove_user_nodes(self, remove_user_nodes: bool):
+        """
+        Setter for the boolean attribute remove_user_nodes
+        True if you want to remove nodes that refer to users when extracting a user profile, False otherwise
+        """
+        self.__remove_user_nodes = remove_user_nodes
 
-    def set_personalized(self, personalized):
-        self.__personalized = personalized
+    @property
+    def remove_items_in_profile(self) -> bool:
+        """
+        Getter for the boolean attribute remove_items_in_profile
+        """
+        return self.__remove_items_in_profile
+
+    @remove_items_in_profile.setter
+    def remove_items_in_profile(self, remove_items_in_profile: bool):
+        """
+        Setter for the boolean attribute remove_items_in_profile
+        True if you want to remove item nodes that are in the user ratings, False otherwise
+        """
+        self.__remove_items_in_profile = remove_items_in_profile
+
+    @property
+    def remove_properties(self) -> bool:
+        """
+        Getter for the boolean attribute remove_properties
+        """
+        return self.__remove_properties
+
+    @remove_properties.setter
+    def remove_properties(self, remove_properties: bool):
+        """
+        Setter for the boolean attribute remove_properties
+        True if you want to remove property nodes, False otherwise
+        """
+        self.__remove_properties = remove_properties
 
     @abstractmethod
     def predict(self, ratings: pd.DataFrame, recs_number: int, items_directory: str,
                 candidate_item_id_list: List = None):
+        """
+        Abstract method that should encapsulate the computation process for the PageRank
+        """
         raise NotImplementedError
 
-    def clean_rank(self, rank: Dict, user_id: str,
```

Comment:
forse è meglio lasciare i parametri qui anziché nel costruttore, in quanto semanticamente quando voglio pulire il rank chiamo questo metodo e mi aspetto di potergli dire cosa eliminare, ma comunque nulla di ché, sottigliezza, per ora puoi lasciare tranquillamente così